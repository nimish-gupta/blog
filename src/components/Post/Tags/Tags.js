// @flow strict
import React from 'react';
import * as Styled from './Tags.style';

type Props = {
	tags: string[],
	tagSlugs: string[],
};

const Tags = ({ tags, tagSlugs }: Props) => (
	<Styled.Tags>
		<Styled.List>
			{tagSlugs &&
				tagSlugs.map((slug, i) => (
					<Styled.Item key={tags[i]}>
						<Styled.ItemLink to={slug}>{tags[i]}</Styled.ItemLink>
					</Styled.Item>
				))}
		</Styled.List>
	</Styled.Tags>
);

export default Tags;
