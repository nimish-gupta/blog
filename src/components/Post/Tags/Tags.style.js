import styled from 'styled-components';
import { Link } from 'gatsby';

import { marginBottom } from '../../../assets/styled-components/mixins';

export const Tags = styled.div`
	${marginBottom(0.5)}
`;

export const List = styled.ul`
	list-style: none;
	margin: 0 -10px;
	padding: 0;
`;
export const Item = styled.li`
	display: inline-block;
	margin: 10px 5px;
`;

export const ItemLink = styled(Link)`
	display: inline-block;
	height: ${(props) => props.theme.button.height};
	padding: 0 24px;
	line-height: ${(props) => props.theme.button.height};
	border: 1px solid ${(props) => props.theme.color.grayBorder};
	text-decoration: none;
	border-radius: ${(props) => props.theme.button.borderRadius};
	color: ${(props) => props.theme.color.base};

	&:hover,
	&:focus {
		color: ${(props) => props.theme.color.primary};
	}
`;
