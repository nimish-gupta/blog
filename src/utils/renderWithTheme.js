import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from '../assets/styled-components/_variables';

export default function renderWithTheme(component) {
	return renderer.create(
		<ThemeProvider theme={theme}>{component}</ThemeProvider>
	);
}
