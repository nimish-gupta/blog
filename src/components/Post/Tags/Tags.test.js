// @flow strict
import React from 'react';
import renderWithTheme from '../../../utils/renderWithTheme';
import Tags from './Tags';

describe('Tags', () => {
	it('renders correctly', () => {
		const props = {
			tags: ['test_0', 'test_1'],
			tagSlugs: ['/test_0', '/test_1'],
		};

		const tree = renderWithTheme(<Tags {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
