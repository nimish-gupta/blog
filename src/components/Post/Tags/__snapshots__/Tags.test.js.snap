// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly 1`] = `
<div
  className="sc-AxjAm jjusOk"
>
  <ul
    className="sc-AxirZ fVbfkj"
  >
    <li
      className="sc-AxiKw eqNqKJ"
    >
      <a
        className="sc-AxhCb eNEMsA"
        href="/test_0"
      >
        test_0
      </a>
    </li>
    <li
      className="sc-AxiKw eqNqKJ"
    >
      <a
        className="sc-AxhCb eNEMsA"
        href="/test_1"
      >
        test_1
      </a>
    </li>
  </ul>
</div>
`;
